src/components/App.spec.js

```diff
@@ -42,9 +42,8 @@ describe('Local Storage', () => {
 
 describe('Wheel', () => {
   it('starts spinning the wheel on spin button click', () => {
-    const SPINNING_PROMPT = '...'
     render(<App />)
     userEvent.click(screen.getByRole('button', { name: /spin/i }))
-    expect(screen.getByTestId('prompt').textContent).toBe(SPINNING_PROMPT)
+    expect(screen.getByTestId('loadingCircles')).toBeVisible()
```

Comment:
Is the wheel only visible after the click on the `spin` button? If it is visible before, then it doesn't really make sense to click on the button as you don't test anything after you clicked, if that makes sense?

Reply:
Thank you for your review! 😊

What I am trying to do here, is test my boolean state hook `mustSpin` that is responsible for allowing the wheel to spin (exactly one time, then the state hook must be set to `false` and reverted to `true` in order to have another spin). The `onClick` in the `spin Button` component sets `mustSpin` to true, which is a condition for displaying the `LoadingCircles` component. When the wheel stops spinning, it triggers a function that sets `mustSpin === false` so the wheel is ready to be spun again. If `mustSpin === false` the `LoadingCircles` component is not rendered. Thus, when the `LoadingCircles` are visible, I know the wheel is actually currently spinning.

I know the test is kind of abstract and obviously needs explaining or deeper knowledge about other components which is problematic in and of itself, but I couldn't think of another way to test a state - or let alone an animation - with testing-library (which is apparently discouraged anyway). I could definitely benefit from some more examples of meaningful, best-practice testing for React components... *nudgenudge* 😜

Reply:
It's actually not that abstract, I just misunderstood! You could maybe test if the `loadingCircles` is `.not().toBeVisible()` before you click on the event (`userEvent.click(screen.getByRole('button', { name: /spin/i }))`). Maybe you will need a `rerender` but I don't think so. 

I will check and see with Jerry whether we can have another workshop about it :)

Reply:
Great, thank you! And I'll add another expectation to the test as you suggested 😁